File: app/src/main/java/com/pleon/buyt/database/converter/DateConverter.java
package com.pleon.buyt.database.converter;

import java.util.Date;

import androidx.room.TypeConverter;

public class DateConverter {

    @TypeConverter
    public static Date fromTimestamp(Long value) {
        return value == null ? null : new Date(value * 1_000); // multiply by 1000 to make it milliseconds
    }

    @TypeConverter
    public static Long toTimestamp(Date date) {
        return date == null ? null : date.getTime() / 1_000; // divide by 1000 to store as seconds
    }
}

File: app/src/main/kotlin/com/pleon/buyt/database/converter/DateConverter.kt
package com.pleon.buyt.database.converter

import androidx.room.TypeConverter
import java.util.*

class DateConverter {

    /**
     * multiply by 1000 to make it in milliseconds
     */
    @TypeConverter
    fun fromTimestamp(value: Long?) = if (value == null) null else Date(value * 1000)

    /**
     * divide by 1000 to store as seconds
     */
    @TypeConverter
    fun toTimestamp(date: Date?) = if (date == null) null else date.time / 1000
}
